doc/Clock commands/clock_managment_hld.md

```diff
@@ -1,92 +1,170 @@
+	  Value is persistent upcon reboot.
+	* Linux timedatectl with set-timezone flag will be called.
+	  e.g. timedatectl set-timezone "Asia/Kolkata"
 
-### Restrictions/Limitations  
 
-### Testing Requirements/Design  
-Explain what kind of unit testing, system testing, regression testing, warmboot/fastboot testing, etc.,
-Ensure that the existing warmboot/fastboot requirements are met. For example, if the current warmboot feature expects maximum of 1 second or zero second data disruption, the same should be met even after the new feature/enhancement is implemented. Explain the same here.
-Example sub-sections for unit test cases and system test cases are given below. 
+2. config clock set-date "<YYYY-MM-DD HH:MM:SS>" (to set time and date)
+	* Command will recieve single string with date-time format "<YYYY-MM-DD HH:MM:SS>"
+	* It will be possible to call command with the following options:
+		1.	Only with a date <YYYY-MM-DD>
+		2. 	Only with time <HH:MM:SS>
+		3.  both date and time "<YYYY-MM-DD HH:MM:SS>"
+	* Command does not need to be stored in DB.
+	* Linux timedatectl with set-time flag will be called.
+	  e.g. timedatectl set-time "2012-10-30 18:17:16"
 
-#### Unit Test cases  
 
-#### System Test cases
+Both set commands will be written directly to Linux (via imedatectl command), and will be activated immediately.
 
-### Open/Action items - if any 
 
-	
-NOTE: All the sections and sub-sections given above are mandatory in the design document. Users can add additional sections/sub-sections if required.
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
+
+Only timezone configuration will be saved.
+Additional field will be added to DEVICE_METADATA table
+
+```
+DEVICE_METADATA :{
+    "timezone": {{string}}
+}
+```
+
+default value: "Etc/UTC"
+
+
+###  3.2 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+####  3.2.1. <a name='Yangmodel'></a>Yang model
+
+
+Adding to existing Yang model of device_metadata (https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-device_metadata.yang)
+
+```
+    container sonic-device_metadata {
+
+        container DEVICE_METADATA {
+
+            description "DEVICE_METADATA part of config_db.json";
+
+            container localhost{
+			...
+				leaf timezone {
+					type string;
+				}
+```
+
+####  3.2.2. <a name='Climodel'></a>CLI
+
+##### Show CLI
+
+```
+root@host:~$ show clock 
+Sun 15 Jan 2023 06:12:08 PM IST
+
+```
+
+##### Config CLI
+
+```
+root@host:~$ config clock set-timezone "<timezone>"
+
+```
+
+```
+root@host:~$ config clock set-date "<YYYY-MM-DD HH:MM:SS>"
+
+```
+
+
+##  4 <a name='TestPlan'></a>Test Plan
+
+###  4.1 <a name='UnitTestcases'></a>Unit Test cases
+
+1. Good flows:<br>
+	a. set timezone<br>
+	b. set date<br>
+	c. set time<br>
+	d. set date & time<br>
+	e. check reboot / upgrade<br>
+
+2. Bad flows:<br>
+	a. set invalid timezone<br>
```

Comment:
How will the user know if the timezone is invalid? Will there be error logs. Is it possible to do some validation at CLI level and throw errors if the timezone is wrong?

Reply:
We will do the validation on CLI level, and throw an error to CLI in case timezone is wrong.

Reply:
in case NTP is enabled:
1. we block set time/date.
2. we allow timezone configuration and change it (that overrides ntp configuration, and changes time accordingly)

Reply:
hence, if we set time/date - it will be overriden by ntp